pexpect/async.py

```diff
@@ -12,22 +12,30 @@ def expect_async(expecter, timeout=None):
     idx = expecter.new_data(previously_read)
     if idx is not None:
         return idx
-
-    transport, pw = yield from asyncio.get_event_loop()\
-        .connect_read_pipe(lambda: PatternWaiter(expecter), expecter.spawn)
-
+    if not expecter.spawn.async_pw_transport:
+        pw = PatternWaiter()
+        pw.set_expecter(expecter)
+        transport, pw = yield from asyncio.get_event_loop()\
+            .connect_read_pipe(lambda: pw, expecter.spawn)
+        expecter.spawn.async_pw_transport = pw, transport
+    else:
+        pw, transport = expecter.spawn.async_pw_transport
+        pw.set_expecter(expecter)
+        transport.resume_reading()
     try:
         return (yield from asyncio.wait_for(pw.fut, timeout))
     except asyncio.TimeoutError as e:
         transport.pause_reading()
         return expecter.timeout(e)
 
+
 class PatternWaiter(asyncio.Protocol):
     transport = None
-    def __init__(self, expecter):
+
+    def set_expecter(self, expecter):
```

Comment:
I think that you can apply this workaround with "no-like" pattern and resolve critical bug. When you got more free time you can make more accurate resolution of this problem. At this time expect() with "async" option totally unusable. Also need to rename async because in python 3.5 async is name of operator.